src/io/flutter/sdk/FlutterSdkUtil.java

```diff
@@ -35,13 +36,25 @@
 
 
 public class FlutterSdkUtil {
+  /** The environment variable to use to tell the flutter tool which app is driving it. */
+  public static final String FLUTTER_ENVIRONMENT = "FLUTTER_ENVIRONMENT";
```

Comment:
Perhaps `FLUTTER_CLIENT`? `FLUTTER_HOST`? This env var is meant to given the flutter cli tool some info about the environment it's running in: IntelliJ, Travis, ...